icc-x-api/icc-accesslog-x-api.ts

```diff
@@ -109,10 +109,41 @@ export class IccAccesslogXApi extends IccAccesslogApi {
       )
   }
 
+  /**
+   * 1. Check whether there is a delegation with 'hcpartyId' or not.
+   * 2. 'fetchHcParty[hcpartyId][1]': is encrypted AES exchange key by RSA public key of him.
+   * 3. Obtain the AES exchange key, by decrypting the previous step value with hcparty private key
+   *      3.1.  KeyPair should be fetch from cache (in jwk)
+   *      3.2.  if it doesn't exist in the cache, it has to be loaded from Browser Local store, and then import it to WebCrypto
+   * 4. Obtain the array of delegations which are delegated to his ID (hcpartyId) in this patient
+   * 5. Decrypt and collect all keys (secretForeignKeys) within delegations of previous step (with obtained AES key of step 4)
+   * 6. Do the REST call to get all helements with (allSecretForeignKeysDelimitedByComa, hcpartyId)
+   *
+   * After these painful steps, you have the helements of the patient.
+   *
+   * @param hcpartyId
+   * @param patient (Promise)
+   * @param keepObsoleteVersions
+   */
+
+  findByUsingPost(hcpartyId: string, patient: models.Patient) {
```

Comment:
It is confusing: some findByUsingPost methods are using `extractDelegationsSFKs` to find the secret keys and some others are using `extractSFKsHierarchyFromDelegations`. 
If you need to get the secret ids from the hierarchy from certain methods, name them accordingly. 
Plus, describe the operation inside the method like: `findByHcPartyAndPatient` instead of `findByUsingPost`, it will be less confusing.